developer_manual/digging_deeper/reference.rst

```diff
@@ -0,0 +1,604 @@
+===================
+Reference providers
+===================
+
+Reference providers are related with two Nextcloud features:
+
+* Link previews
+* The Smart Picker
+
+Link previews were introduced in Nextcloud 25.
+Apps can register a reference provider to add preview support for some HTTP links.
+To provide link previews, a reference provider needs to:
+
+* Resolve the links (get information about the links)
+* Render links (show this information in the user interface)
+
+The Smart Picker was introduced in Nextcloud 26. This is a user interface component
+allowing users to search or generate links from various places in Nextcloud like Text,
+Talk, Collectives, Notes, Mail...
+Reference providers can be implemented so they appear in the Smart Picker's provider list.
+The Smart Picker can use 2 types of providers:
+
+* the ones to search (using existing Unified Search providers)
+* the ones implementing their own custom picker component
+
+In summary, reference providers can be registered by apps to
+
+* add support for new kinds of HTTP links
+    * resolve the links, get information on the link targets
+    * optionally provide their own reference widgets to have a custom preview rendering
+* extend the Smart Picker
+    * use existing Unified Search providers
+    * or optionally register custom picker components to have a specific user interface
```

Comment:
does this nested list render as expected? I think there has to be a blank line between the first level items

Reply:
![image](https://user-images.githubusercontent.com/11291457/232716003-2231aa1b-c444-4ecb-add9-ce35d87b7997.png)

It's fine without blank lines.

Reply:
okay, I'm sorry. for the nested list I wrote the other day sphinx wouldn't render correctly without the line :shrug: